app/components/telegram-auth.tsx

```diff
@@ -141,6 +192,45 @@ export default function TelegramAuth() {
 		setCodeHash(phoneCodeHash)
 	}
 
+	const handle2FAPasswordSubmit = async () => {
+		try {
+			setLoading(true)
+			setError(undefined)
+
+			if (!srp) {
+				throw new Error('missing secure remote password')
+			}
+
+			if (!client.connected) {
+				await client.connect()
+			}
+			const result = await client.invoke(new Api.auth.CheckPassword({
+				password: await computeCheck(srp, password)
+			}))
+			if (result instanceof Api.auth.AuthorizationSignUpRequired) {
+				throw new Error('user needs to sign up')
+			}
+			if (process.env.NODE_ENV === 'production') {
+				if (result.user.id.value !== BigInt(user?.id ?? 0)) {
```

Comment:
I have one concern with this check: when verifying this on the chat visualization side, I noticed that the user ID from GramJS (TDLib API) is different from the one provided by `@telegram-apps` (bot API) in `initData.user`. I’m guessing that’s because they come from different contexts. It might be worth doing an extra check here for that

Reply:
I think that's because when testing you start the app with your test user (in test mode) but then login as a production user. Hence the check here only being activated in production. Alternatively we could fix GramJS to use the test server so you can login with your test account and always have this check enabled.

Reply:
ahh that makes sense! thanks for that
I took a look and apparently there're some test accounts we can use to login. I'm going to take a better look into it